modules/custom/activity_send/modules/activity_send_email/src/Plugin/AdvancedQueue/JobType/ActivitySendEmailJobType.php

```diff
@@ -0,0 +1,344 @@
+          $message = $message_storage->load($activity->getFieldValue('field_activity_message', 'target_id'));
+          $message_template_id = $message->getTemplate()->id();
+
+          // Prepare an array of all details required to process the item.
+          $parameters = [
+            'activity' => $activity,
+            'message' => $message,
+            'message_template_id' => $message_template_id,
+          ];
+
+          // We want to give preference to users who have set notification
+          // settings as 'immediately'.
+          $email_frequencies = [
+            'immediately',
+            'daily',
+            'weekly',
+            'none',
+          ];
+
+          // Let's store the users IDs which will be processed by the loop.
+          $processed_users = [];
+          foreach ($email_frequencies as $email_frequency) {
+            // Get the 'target recipients' of who have their 'email notification
+            // preference' matching to current $email_frequency.
+            if ($target_recipients = EmailActivityDestination::getSendEmailUsersIdsByFrequency($recipients, $message_template_id, $email_frequency)) {
+              // Update process users.
+              $processed_users = array_merge($processed_users, $target_recipients);
+
+              // We load all the target accounts.
+              $parameters['target_recipients'] = $target_recipients;
+              // We set the frequency of email.
+              $parameters['frequency'] = $email_frequency;
+
+              // If the batch has langcode.
+              if (!empty($data['langcode'])) {
+                $parameters['langcode'] = $data['langcode'];
+              }
+
+              // Send for further processing.
+              $this->sendToFrequencyManager($parameters);
+            }
+          }
+
+          // There is possibility where the users have not saved their
+          // 'email notification preferences'. So, we check the difference
+          // between the processed user IDs and original recipients users IDs
+          // and send emails according to default 'frequency' set by site
+          // manager. If SM has also not set, take 'immediately' as frequency.
+          if ($remaining_users = array_diff($recipients, $processed_users)) {
+            // Grab the platform default "Email notification frequencies".
+            $template_frequencies = $this->swiftmailSettings->get('template_frequencies') ?: [];
+            // Determine email frequency to use, defaults to immediately.
+            $parameters['frequency'] = $template_frequencies[$message_template_id] ?? FREQUENCY_IMMEDIATELY;
+            $parameters['target_recipients'] = $remaining_users;
+            $this->sendToFrequencyManager($parameters);
+          }
+        }
+      }
+
+      // By default mark the Job as failed.
+      $this->getLogger('activity_send_email_worker')->error('The job was finished correctly.');
```

Comment:
![image](https://user-images.githubusercontent.com/8435994/119806343-beaed980-beff-11eb-8bf9-10e9d68ae8f1.png)

This should be changed to successful I believe.

Reply:
well spotted haha